.github/workflows/ccpp.yml

```diff
@@ -46,7 +50,15 @@ jobs:
     - name: Build
       run: |
            cd gccrs-build; \
-           make -j $(nproc)
+           make -j $(nproc) > log 2>&1
+           
+    - name: Build logs
+      run: cat gccrs-build/log
+
+    - name: Check for new warnings
+      run: |
+           grep 'warning:' gccrs-build/log | sort > log_warnings;
```

Comment:
This *step* will fail in case that `grep` doesn't find a single *warning*.  We shall defer that latent problem until relevant...  ;-)